clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/InfrastructureIT.java

```diff
@@ -838,4 +838,484 @@ public void shouldGetRawResources() throws InterruptedException, IOException {
+        KubeTestUtils.loadJson("classpath:requests/deploy_manifest.json")
+            .withValue("deployManifest.account", ACCOUNT1_NAME)
+            .withValue("deployManifest.moniker.app", appName)
+            .withValue("deployManifest.manifests", List.of(networkPolicy))
+            .asList();
+
+    KubeTestUtils.deployAndWaitStable(baseUrl(), body, ns, "networkPolicy" + " " + name);
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          // ------------------------- when --------------------------
+          System.out.println("> Sending get securityGroups request");
+          Response resp =
+              get(
+                  baseUrl()
+                      + "/securityGroups/"
+                      + ACCOUNT1_NAME
+                      + "/kubernetes/"
+                      + ns
+                      + "/networkolicy "
+                      + name);
+
+          // ------------------------- then --------------------------
+          System.out.println(resp.asString());
+          if (resp.statusCode() == 404) {
+            return false;
+          }
+          resp.then().statusCode(200);
+          var displayName = resp.jsonPath().getString("displayName");
+          return displayName != null && displayName.equals(name);
+        },
+        CACHE_TIMEOUT_MIN,
+        TimeUnit.MINUTES,
+        "Waited "
+            + CACHE_TIMEOUT_MIN
+            + " minutes for GET /securityGroups/"
+            + ACCOUNT1_NAME
+            + "/kubernetes/"
+            + ns
+            + "/networkolicy "
+            + name
+            + " to return valid data");
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given a kubernetes deployment\n"
+          + "When sending get clusters /applications/{appName}/serverGroupManagers\n"
+          + "Then the deployment should be present in serverGroups list of the response\n===")
+  @Test
+  public void shouldGetServerGroupManagerForApplication() throws InterruptedException, IOException {
+    // ------------------------- given --------------------------
+    String ns = kubeCluster.getAvailableNamespace();
+    String deploymentName = "inframyapp";
+    String appName = "getclusters";
+    System.out.println("> Using namespace " + ns);
+    KubeTestUtils.deployAndWaitStable(
+        baseUrl(), ACCOUNT1_NAME, ns, "deployment", deploymentName, appName);
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          // ------------------------- when --------------------------
+          System.out.println("> Sending get clusters request");
+          Response resp = get(baseUrl() + "/applications/" + appName + "/serverGroupManagers");
+
+          // ------------------------- then --------------------------
+          System.out.println(resp.asString());
+          if (resp.statusCode() == 404) {
+            return false;
+          }
+          resp.then().statusCode(200);
+          List serverGroupList = resp.jsonPath().getList("[0].serverGroups");
```

Comment:
Would this fail with an unwanted error if the returned list is empty (i.e. there's no index `0`)? Probably it's better to check the root list instead of the nested `serverGroups`.